oak_functions_launcher/tests/integration_test.rs

```diff
@@ -113,6 +115,8 @@ async fn test_launcher_echo_virtual() {
 
     let response = client.invoke(b"xxxyyyzzz").await.expect("failed to invoke");
     assert_eq!(std::str::from_utf8(&response).unwrap(), "xxxyyyzzz");
+
+    background.kill();
```

Comment:
Could we have this happen on Drop, instead of explicitly having to call it?

Reply:
Nice idea. Curiously the function that returns this [claims that this already happens](https://github.com/project-oak/oak/blob/ef40f815281d4a3d6bc7959eb9797efa51d02e21/xtask/src/testing.rs#L39).

We cannot implement this on the existing type `Box<dyn Running>` since it conflicts with the existing Drop implementation of Box. We could however accomplish this by returning a type of our own, similar to `tokio::process::Command`. Might be worth doing so, but it'd be a slightly larger of xtask.

Reply:
Went with a small solution that adds an extra struct for background tasks that implements the dropping. Ideally I think _all_ tasks within xtask should be killed on drop. However implementing that may not be possible. The current solution will do for now.